bftengine/include/bftengine/ReplicaConfig.hpp

```diff
@@ -147,6 +148,21 @@ class ReplicaConfig : public concord::serialize::SerializableFactory<ReplicaConf
   CONFIG_PARAM(keyExchangeOnStart, bool, false, "whether to perform initial key exchange");
   CONFIG_PARAM(keyViewFilePath, std::string, ".", "TODO");
 
+  // Time Service
+  CONFIG_PARAM(timeServiceEnabled, bool, false, "whether time service enabled");
+  CONFIG_PARAM(timeServiceSoftLimitMillis,
+               std::chrono::milliseconds,
+               std::chrono::milliseconds{500},
+               "if delta between primary's time is greater than the soft limit, a warning log message to be printed");
+  CONFIG_PARAM(timeServiceHardLimitMillis,
+               std::chrono::milliseconds,
+               std::chrono::seconds{1} * 3,
+               "if delta between primary's time is greater than the hard limit, the consesnus does not proceed");
+  CONFIG_PARAM(timeServiceEpsilonMillis,
+               std::chrono::milliseconds,
+               std::chrono::milliseconds{1},
+               "time provided to execution is max(consensus_time, last_time + timeServiceEpsilonMillis)");
+
```

Comment:
You need to add new parameters to inline std::ostream& operator<<(std::ostream& os, const ReplicaConfig& rc), serializeDataMembers/deserializeDataMembers function, too

Reply:
@yuliasherman , updated